Assets/Mirror/Runtime/NetworkIdentity.cs

```diff
@@ -161,11 +161,9 @@ public sealed class NetworkIdentity : MonoBehaviour
         /// </summary>
         public bool hasAuthority { get; internal set; }
 
-        /// <summary>
-        /// The set of network connections (players) that can see this object.
-        /// <para>null until OnStartServer was called. this is necessary for SendTo* to work properly in server-only mode.</para>
-        /// </summary>
-        public Dictionary<int, NetworkConnection> observers;
+        // interest management
+        internal HashSet<NetworkConnectionToClient> rebuild = new HashSet<NetworkConnectionToClient>();
+        internal List<NetworkConnectionToClient> observers = new List<NetworkConnectionToClient>();
```

Comment:
Turning this dictionary into a list should be it's own PR.

I proposed this a long time ago,  it wasn't accepted then.

I would suggest using a HashSet here instead,  because we need to call Contains all the times,  and Contains is O(n) in lists, but O(1) in sets

Reply:
was already changed to hashset in 'x' branch.
the pr was just a preview to show it could be done. it's outdated already:)